stubs/parsimonious/parsimonious/grammar.pyi

```diff
@@ -0,0 +1,69 @@
+from collections import OrderedDict
+from typing import Any, Callable, Mapping, NoReturn
+from typing_extensions import Self
+
+from parsimonious.exceptions import BadGrammar as BadGrammar, UndefinedLabel as UndefinedLabel
+from parsimonious.expressions import (
+    _CALLABLE_TYPE as _CALLABLE_TYPE,
+    Expression as Expression,
+    Literal as Literal,
+    Lookahead as Lookahead,
+    Not as Not,
+    OneOf as OneOf,
+    OneOrMore as OneOrMore,
+    Optional as Optional,
+    Regex as Regex,
+    Sequence as Sequence,
+    TokenMatcher as TokenMatcher,
+    ZeroOrMore as ZeroOrMore,
+    expression as expression,
+    is_callable as is_callable,
+)
+from parsimonious.nodes import Node as Node, NodeVisitor as NodeVisitor
+from parsimonious.utils import evaluate_string as evaluate_string
+from six import text_type
+
+class Grammar(OrderedDict[str, Expression]):
+    default_rule: Expression | Any
+    def __init__(self, rules: str = ..., **more_rules: Expression | _CALLABLE_TYPE) -> None: ...
+    def default(self, rule_name: str) -> Grammar: ...
+    def parse(self, text: str, pos: int = ...) -> Node: ...
+    def match(self, text: str, pos: int = ...) -> Node: ...
+
+class TokenGrammar(Grammar): ...
+class BootstrappingGrammar(Grammar): ...
+
+rule_syntax: str
+
+class LazyReference(text_type):
+    name: str
+
+class RuleVisitor(NodeVisitor):
+    quantifier_classes: dict[str, Expression]
+    visit_expression: Callable[[Self, Node, list[Any]], Any]
+    visit_term: Callable[[Self, Node, list[Any]], Any]
+    visit_atom: Callable[[Self, Node, list[Any]], Any]
+    custom_rules: dict[str, Expression]
```

Comment:
Also maybe should have `type[]`?

Reply:
So this is used to updated the rule_map which contains instantiated Expressions so think this is correct as is?

https://github.com/erikrose/parsimonious/blob/3da7e804c07d4e495873be208701b5c955247c58/parsimonious/grammar.py#L438-L447